src/signals/incident/components/form/Header/index.js

```diff
@@ -1,58 +1,80 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-
+import { FormattedMessage } from 'react-intl';
 import { Validators } from 'react-reactive-form';
+import get from 'lodash.get';
 
 import './style.scss';
+import flattenObject from '../../../services/object-flatten';
 
-const Header = ({ meta, options, touched, hasError, getError, children }) => (
-  <div className={`header ${touched && (hasError('required') || hasError('email') || hasError('maxLength') || hasError('custom')) ? 'header--invalid' : ''}`}>
-    <div className="header__label">{meta && meta.label}
-      {(meta.label && (!options || !options.validators)) || (options && options.validators && !options.validators.includes(Validators.required)) ?
-        <span className="header--not-required">(optioneel)</span>
-      : ''}
-    </div>
-    <div className="header__subtitle">{meta && meta.subtitle}</div>
 
-    <div className="header__errors">
-      <div className="header__errors__item">
-        {touched
-        && hasError('required')
-        && 'Dit is een verplicht veld'}
-      </div>
+const Header = ({ meta, options, touched, hasError, getError, children, parent }) => {
+  const {
+    label,
+    subtitle,
+  } = meta;
 
-      <div className="header__errors__item">
-        {touched
-        && hasError('email')
-        && 'Het moet een geldig e-mailadres zijn'}
-      </div>
+  const incident = get(parent, 'meta.incident', {});
+  const shallowValues = flattenObject(incident, '', ':');
+  const isLabelDescriptor = label !== null && typeof label === 'object';
```

Comment:
With the types listed below, the condition will also be truthy. Is that intentional?

```jsx
typeof {a: 1} === 'object';
typeof [1, 2, 4] === 'object';
typeof new Date() === 'object';
typeof /regex/ === 'object';
typeof new Boolean(true) === 'object'; 
typeof new Number(1) === 'object'; 
typeof new String('abc') === 'object';
```

Reply:
The idea is that the input is either a LabelDescriptor or a simple string. The check here is simply to distinguish between those two. So i think the current code satisfies that requirement.

However, clearly it's unclear what kind of object it is. So clarified the prop type a bit.